### PR TITLE
feat: support query manager

### LINKED ISSUE #291: support for assigning an id to each query and pass it to the store
support for assigning an id to each query and pass it to the store

### LINKED ISSUE #292: support show queries
```
> show queries

qid      query      database  duration   status  host
---      -----      -------   --------   -----   -----
100001   select...  db0       123us      running 127.0.0.4:8088 
```

### LINKED ISSUE #293: support kill queries
```
> kill query 1000001
```

### PR DESCRIPTION
**Pull request by wtsclwq**
### What problem does this PR solve?

Issue Number: close #291 , close #292, close #293  

Discussion Number #285
### What is changed and how it works?

In previous versions, each **sql node** assigned a query id from 0, which is **not unique** under a distributed cluster. In this pull request, all sql nodes can assign a **globally unique id** to each query statement they receive  and support the `SHOW QUERIES` statement under the distributed cluster to display the detailed letter of the all running SQL. Then we can use the `KILL QUERY` statement  to abort the running statement on the store nodes.

### How Has This Been Tested?
Simulate some slow queries, and then use `SHOW QUERIES` and `KILL QUERY` statements on them.

image

- [ ] Test A
- [ ] Test B
- [x] Test cases to be added
- [ ] No code

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open_src/influx/influxql/internal/internal.pb.go [hunk 1]
```diff
@@ -137,11 +137,6 @@ func (m *Measurement) GetEngineType() uint32 {
 	return 0
 }
 
-func init() {
-	proto.RegisterType((*Measurements)(nil), "influxql.Measurements")
-	proto.RegisterType((*Measurement)(nil), "influxql.Measurement")
```
Comment (shilinlee) [inline open_src/influx/influxql/internal/internal.pb.go:142]: Here is the generated code.

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: lib/metaclient/meta_callback.go
Comment (shilinlee) [inline lib/metaclient/meta_callback.go:1]: Please add some unit tests.

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: app/ts-store/transport/handler/handlers_process.go [hunk 2]
```diff
@@ -117,6 +119,50 @@ func (h *ShowTagValuesCardinality) Process() (codec.BinaryCodec, error) {
...
+			isExist = true
+			manager.Abort(qid)
+			isAborted = manager.Aborted(qid)
+		}
+	}
+	query.VisitManagers(killQueryByIDFn)
+	err := &errno.Error{}
+	if !isExist {
+		err = err.SetErrno(errno.ErrQueryNotFound)
+		err.SetMessage(fmt.Sprintf("the query %d is not running on this store node", qid))
```
Comment (shilinlee) [inline app/ts-store/transport/handler/handlers_process.go:156]: 1. define error code in `lib/errno/code.go`
2. define error message in `lib/errno/message.go`
3. e.g. `errno.NewError(errno.ErrQueryNotFound, qid)`

File: app/ts-store/transport/handler/handlers_process.go [hunk 3]
```diff
@@ -117,6 +119,50 @@ func (h *ShowTagValuesCardinality) Process() (codec.BinaryCodec, error) {
...
+		h.rsp.Err = netstorage.MarshalError(err)
+	} else if !isAborted {
+		err = err.SetErrno(errno.ErrKillQueryFail)
```
Comment (shilinlee) [inline app/ts-store/transport/handler/handlers_process.go:159]: The same as above.

File: app/ts-store/transport/handler/handlers_process.go [hunk 4]
```diff
@@ -117,6 +119,50 @@ func (h *ShowTagValuesCardinality) Process() (codec.BinaryCodec, error) {
...
+	rsp.QueryExeInfos = queries
+
+	return rsp, nil
+
+}
+
+func (h *KillQuery) Process() (codec.BinaryCodec, error) {
+	qid := h.req.GetQueryID()
+	isExist := false
+	isAborted := false
```
Comment (shilinlee) [inline app/ts-store/transport/handler/handlers_process.go:140]:
```suggestion
	var isExist bool
	var isAborted bool
```

File: lib/errno/code.go [hunk 5]
```diff
@@ -100,6 +100,8 @@ const (
 	NoColValToColumnFunc         = 1119
 	InvalidQuerySchema           = 1120
 	InvalidQueryStat             = 1121
+	ErrQueryNotFound             = 1122
+	ErrKillQueryFail             = 1123
```
Comment (shilinlee) [inline lib/errno/code.go:104]: add error message in `lib/errno/message.go`

File: open_src/influx/meta/errors.go [hunk 6]
```diff
@@ -61,10 +61,12 @@ var (
...
-	ErrInvalidPtView     = errors.New("invalid ptView number")
-	ErrDataViewBootStrap = errors.New("cluster is bootstrapping for initial data view")
-	ErrDuplicateShardKey = errors.New("duplicate shard key")
-	ErrInvalidShardKey   = errors.New("invalid shard key")
+	ErrInvalidPtView        = errors.New("invalid ptView number")
+	ErrDataViewBootStrap    = errors.New("cluster is bootstrapping for initial data view")
+	ErrDuplicateShardKey    = errors.New("duplicate shard key")
+	ErrInvalidShardKey      = errors.New("invalid shard key")
+	ErrKillQueryFail        = errors.New("failed to kill query on %s")
+	ErrKillNotExistentQuery = errors.New("the query %d is not running on any store node")
```
Comment (shilinlee) [inline open_src/influx/meta/errors.go:69]:
```suggestion
```

move to `lib/errno/message.go`

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: open_src/influx/query/executor.go [hunk 7]
```diff
@@ -51,6 +52,8 @@ var (
 
 	// ErrAlreadyKilled is returned when attempting to kill a query that has already been killed.
 	ErrAlreadyKilled = errors.New("already killed")
+
+	ErrQueryIDOffsetNotInit = errors.New("query id offset has not been initialized, please try again")
```
Comment (shilinlee) [inline open_src/influx/query/executor.go:56]: unused code

```suggestion
```

File: lib/errno/message.go [hunk 8]
```diff
@@ -132,6 +132,8 @@ var messageMap = map[Errno]*Message{
 	NoColValToColumnFunc:         newWarnMessage("no func to transform colval into column: %s", ModuleQueryEngine),
 	InvalidQuerySchema:           newWarnMessage("invalid query schema", ModuleQueryEngine),
 	InvalidQueryStat:             newWarnMessage("invalid query stat", ModuleQueryEngine),
+	ErrQueryNotFound:             newWarnMessage("query %d is not running on this store node", ModuleQueryEngine),
```
Comment (shilinlee) [inline lib/errno/message.go:135]:
```suggestion
	ErrQueryNotFound:             newWarnMessage("no such query id: %d", ModuleQueryEngine),
```

**Review by shilinlee — CHANGES_REQUESTED** [review]

**Review by shilinlee — CHANGES_REQUESTED** [review]

**Review by shilinlee — APPROVED** [review]

**Review by scuzyj — APPROVED** [review]

**Pull request merged**